From 86ea261f67ce9563fbc52fef41e61fb8532fc979 Mon Sep 17 00:00:00 2001
From: Pankaj Kumar <pankaj.kr@huawei.com>
Date: Thu, 28 Sep 2017 20:41:28 +0800
Subject: [PATCH 436/444] HBASE-18890, Backport HBASE-14499 (Master
 coprocessors shutdown will not happen on master
 abort) to branch-1

Signed-off-by: tedyu <yuzhihong@gmail.com>
(cherry picked from commit b2943504b867e7b26e7c6433f026a602ef42a0ab)

Change-Id: I24b39a0e914abefa1632bde8368689452463252c
Author: Pankaj Kumar
Reason: Bug
Ref: CDH-61618
---
 .../org/apache/hadoop/hbase/master/HMaster.java    |    7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java
index ecadcbc..454d3ee 100644
--- a/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java
+++ b/hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java
@@ -2300,7 +2300,12 @@ public class HMaster extends HRegionServer implements MasterServices, Server {
           getLoadedCoprocessors());
     }
     if (t != null) LOG.fatal(msg, t);
-    stop(msg);
+
+    try {
+      stopMaster();
+    } catch (IOException e) {
+      LOG.error("Exception occurred while stopping master", e);
+    }
   }
 
   @Override
-- 
1.7.9.5

